internal/character/clara/talent.go

```diff
@@ -0,0 +1,153 @@
+package clara
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	TalentCounter key.Modifier = "clara-talent-counter"
+	TalentMark    key.Modifier = "clara-talent-mark"    // MAvatar_Klara_00_PassiveATK_Mark for BPSkill_Revenge
+	TalentRes     key.Modifier = "clara-talent-dmg-res" // MAvatar_Klara_00_Passive_DamageReduce
+)
+
+// Under the protection of Svarog, DMG taken by Clara when hit by enemy
+// attacks is reduced by 10%. Svarog will mark enemies who attack Clara with
+// his Mark of Counter and retaliate with a Counter, dealing Physical DMG
+// equal to *% of Clara's ATK.
+//
+// flow:
+//
+// onBeforeBeingAttacked: + TalentMark, + TalentRes
+//
+// onAfterAttack: + follow-up attack
+func init() {
+	modifier.Register(TalentCounter, modifier.Config{
+		Listeners: modifier.Listeners{
+			// add marker modifier on attacker enemy
+			OnBeforeBeingAttacked: func(mod *modifier.Instance, e event.AttackStart) {
+				mod.Engine().AddModifier(e.Attacker, info.Modifier{
+					Name:   TalentMark,
+					Source: mod.Source(),
+				})
+			},
+
+			// remove marker modifier on all enemies when clara dies
+			OnBeforeDying: func(mod *modifier.Instance) {
+				for _, enemyID := range mod.Engine().Enemies() {
+					mod.Engine().RemoveModifier(enemyID, TalentMark)
+				}
+			},
+		},
+	})
```

Comment:
I don't think `TalentCounter` modifier is necessary:
* The `TalentMark` can be added via `talentActionEndListener` for the Clara hit case
* Since the `TalentMark` has no logic or stats, we don't need to remove it if Clara dies